storage/mysql/log_storage_test.go

```diff
@@ -176,32 +178,28 @@ func TestQueueDuplicateLeaf(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.desc, func(t *testing.T) {
-			runLogTX(s, tree, t, func(ctx context.Context, tx storage.LogTreeTX) error {
-				existing, err := tx.QueueLeaves(ctx, test.leaves, fakeQueueTime)
-				if err != nil {
-					t.Errorf("Failed to queue leaves: %v", err)
-					return err
-				}
+			existing, err := s.QueueLeaves(ctx, tree, test.leaves, fakeQueueTime)
+			if err != nil {
+				t.Fatalf("Failed to queue leaves: %v", err)
+			}
 
-				if len(existing) != len(test.want) {
-					t.Fatalf("|QueueLeaves()|=%d; want %d", len(existing), len(test.want))
-				}
-				for i, want := range test.want {
-					got := existing[i]
-					if want == nil {
-						if got != nil {
-							t.Fatalf("QueueLeaves()[%d]=%v; want nil", i, got)
-						}
-						return nil
-					}
-					if got == nil {
-						t.Fatalf("QueueLeaves()[%d]=nil; want non-nil", i)
-					} else if !bytes.Equal(got.LeafIdentityHash, want.LeafIdentityHash) {
-						t.Fatalf("QueueLeaves()[%d].LeafIdentityHash=%x; want %x", i, got.LeafIdentityHash, want.LeafIdentityHash)
+			if len(existing) != len(test.want) {
+				t.Fatalf("|QueueLeaves()|=%d; want %d", len(existing), len(test.want))
+			}
+			for i, want := range test.want {
+				got := existing[i]
+				if want == nil {
+					if got, want := status.FromProto(got.GetStatus()).Code(), codes.OK; got != want {
```

Comment:
Would `gotStatus`, `wantStatus` to avoid shadowing variables in the enclosing scope be more readable?

Reply:
Discovered today that this can be simplified to `got.Status != nil`